hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -1830,14 +1837,23 @@ private OzoneFileStatus getOzoneFileStatus(String volumeName,
       // Check if the key is a file.
       String fileKeyBytes = metadataManager.getOzoneKey(
               volumeName, bucketName, keyName);
+      // if the queried key is already a directory key (For eg: Keyname= a/b/c/)
+      if (fileKeyBytes.endsWith(OZONE_URI_DELIMITER)){
```

Comment:
Should we need this additional check for the trailing slash? Will the below line fetch the given directory a/b/c/ or file a/b/c key from KeyTable, right?
`fileKeyInfo = metadataManager.getKeyTable().get(fileKeyBytes);`

Reply:
Removed this check.